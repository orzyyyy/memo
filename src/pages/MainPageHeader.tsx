import React, { useState } from 'react';
import { MainPageProps } from './MainPage';
import Button from '../component/Button';
import Input from '../component/Input';
import Select from '../component/Select';

const MainPageHeader = ({
  onExhentaiDownload,
  onEdit,
  exhentaiDateSet,
  onExhentaiSelectChange,
  onExhentaiLoadList,
}: MainPageProps) => {
  const [selectValue, setSelectValue] = useState();

  const handleSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const value = e.target.value;
    onExhentaiSelectChange(value);
    setSelectValue(value);
  };

  const hanldeKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    const value = (e.target as any).value;
    if (e.key === 'Enter' && value) {
      onExhentaiDownload(value);
    }
  };

  return (
    <header style={{ height: 48, display: 'flex', alignItems: 'center', marginLeft: 16, marginTop: 8 }}>
      <Select
        style={{ width: 140 }}
        value={selectValue || (exhentaiDateSet.length ? exhentaiDateSet[0] : '')}
        onChange={handleSelectChange}
      >
        {exhentaiDateSet.map(timeStamp => (
          <option value={timeStamp} key={`exhentai-time-stamp-${timeStamp}`} style={{ height: 40 }}>
            {timeStamp}
          </option>
        ))}
      </Select>
      <Input onKeyDown={hanldeKeyDown} style={{ width: 280 }} />
      <Button onClick={onExhentaiLoadList}>列表</Button>
      <Button onClick={(e: React.MouseEvent) => onEdit(undefined, true, { x: e.pageX, y: e.pageY })}>+</Button>
    </header>
  );
};

export default MainPageHeader;
