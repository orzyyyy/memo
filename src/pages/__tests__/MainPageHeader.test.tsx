import React from 'react';
import { shallow, mount } from 'enzyme';
import MainPageHeader from '../MainPageHeader';

describe('MainPageHeader', () => {
  const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  afterAll(() => {
    errorSpy.mockRestore();
  });

  it('render correctly', () => {
    const onEdit = jest.fn();
    const onExhentaiDownload = jest.fn();
    const onExhentaiSelectChange = jest.fn();
    const onExhentaiLoadList = jest.fn();
    const wrapper = shallow(
      <MainPageHeader
        onEdit={onEdit}
        onExhentaiDownload={onExhentaiDownload}
        onExhentaiSelectChange={onExhentaiSelectChange}
        onExhentaiLoadList={onExhentaiLoadList}
        exhentaiDateSet={[]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('onEdit of init button works correctly', () => {
    const onEdit = jest.fn();
    const onExhentaiDownload = jest.fn();
    const onExhentaiSelectChange = jest.fn();
    const onExhentaiLoadList = jest.fn();
    const wrapper: any = mount(
      <MainPageHeader
        onEdit={onEdit}
        onExhentaiDownload={onExhentaiDownload}
        onExhentaiSelectChange={onExhentaiSelectChange}
        onExhentaiLoadList={onExhentaiLoadList}
        exhentaiDateSet={[]}
      />,
    );
    wrapper
      .find('Button')
      .at(1)
      .props()
      .onClick();
    expect(onEdit).toHaveBeenCalled();
  });
});
