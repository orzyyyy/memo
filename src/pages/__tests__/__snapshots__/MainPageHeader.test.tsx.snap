// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageHeader render correctly 1`] = `
<header
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 48,
      "marginLeft": 16,
      "marginTop": 8,
    }
  }
>
  <Select
    onChange={[Function]}
    style={
      Object {
        "width": 140,
      }
    }
    value=""
  />
  <Input
    onKeyDown={[Function]}
    style={
      Object {
        "width": 280,
      }
    }
  />
  <Button
    onClick={[MockFunction]}
  >
    列表
  </Button>
  <Button
    onClick={[Function]}
  >
    +
  </Button>
</header>
`;
